File: svt-projekat2024/src/main/java/rs/ac/uns/ftn/svt/model/Image.java
package rs.ac.uns.ftn.svt.model;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
public class Image {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String path;

    @OneToOne
    @JoinColumn(name = "user_id", unique = true)
    private User user;

    @ManyToOne
    @JoinColumn(name = "facility_id", nullable = false)
    private Facility facility;

}

File: kvt-projekat2024/src/app/models/Image.ts
import { User } from "./User";
import { Facility } from "./Facility";

export class Image {
  id: number;
  path: string;
  user: User;
  facility: Facility;

  constructor(
    id: number,
    path: string,
    user: User,
    facility: Facility
  ) {
    this.id = id;
    this.path = path;
    this.user = user;
    this.facility = facility;
  }
}
